lib/phoenix/socket.ex

```diff
@@ -178,7 +179,8 @@ defmodule Phoenix.Socket do
             serializer: nil,
             topic: nil,
             transport: nil,
-            transport_pid: nil
+            transport_pid: nil,
+            meta: %{}
```

Comment:
I don't want to make the Socket "fatter". Can we please not add this field to the socket? :)

Reply:
I would prefer if we passed this as an argument to `connect` and let the user extract what they need. There is no reason to keep all of this information around. Although I am not sure what you and Chris discussed. In my my mind, it is ok to call `connect/3` with the meta if it is available, otherwise fallback to `connect/2`.

Reply:
I believe that the discussion was on the fence between the two choices. If I understand your proposal, we would call connect/3 if the function has been exported.

Reply:
> If I understand your proposal, we would call connect/3 if the function has been exported.

Yes. @chrismccord, thoughts?

Other than this everything in this PR is perfect. I like the three options: :uri, :x_headers and :peer_data :+1:.